annotation_agreement/aggregate_annotations.py

```diff
@@ -0,0 +1,216 @@
+"""
+This script is used to generate the aggregated annotations.
+Usage:
+1) download annotation csv (aggregate_annotations.csv)
+   and list of files (videofiles.txt)
+2) run ./data_prep_linux (or mac depending on your OS)
+3) run python3 aggregate_annotations.py
+4) aggregated.csv is the final annotation file containing aggregated annotations
+   and respective confidences. plots folder contains respective plots
+"""
+
+import shutil
+import os
+import numpy
+import pandas as pd
+import matplotlib.pyplot as plt
+import argparse
+
+def aggregate_annotations(file,class_number):
+
+    data = pd.read_csv(file)
+
+    # Create Dataframe
+    aggregation = {'Sample_Name': [],
+                   'Winner_annotation': [],
+                   'Confidence': [],
+                   'Number_annotations': []}
+    
+    df = pd.DataFrame(aggregation, columns=list(aggregation.keys()))
+
+
+    if class_number==1:
+        num_anot = (pd.crosstab(data.Sample_name, data.Class1))
+        conf = (pd.crosstab(data.Sample_name, data.Class1))
+    elif class_number==2:
+        num_anot = (pd.crosstab(data.Sample_name, data.Class2))
+        conf = (pd.crosstab(data.Sample_name, data.Class2))
+    else:
+        num_anot = (pd.crosstab(data.Sample_name, data.Class3))
+        conf = (pd.crosstab(data.Sample_name, data.Class3))
+
+    
+    # Number_annotations
+    num_anot['sum'] = num_anot.sum(axis=1)
+    num_anot = num_anot.reset_index()
+
+    df['Number_annotations'] = num_anot['sum']
+       
+    # Confidence
+    res = conf.div(conf.sum(axis=1), axis=0)*100
+    res = res.reset_index()
+
+    # Values to Dataframe
+    df['Sample_Name'] = res['Sample_name']
+    sav=res['Sample_name']
+    res=res.drop(['Sample_name'], axis=1)
+    res['Max'] = res.idxmax(axis=1)
+    res['max_value'] = res.max(axis=1)
+
+    df['Winner_annotation'] = res["Max"]
+    df['Confidence'] = res["max_value"]
+
+    return df
+
+
+def save_to_csv(df,name):   
+
+    df.to_csv(name, index=False)
+
+
+def report_annotations(file, class_number,annotators):
+
+    data = pd.read_csv(file)
+    df = aggregate_annotations(file,class_number)
+    csv_name = 'aggregate_annotations_of_class' + str(class_number) + '.csv'
+    save_to_csv(df, csv_name)
+
+    class_column = 'Class' + str(class_number)
+
+    with open('audiofiles.txt') as f:
```

Comment:
I think @sofiaele that this file is no more needed right? I mean the audio files are now taken as a unique set. If yes, please remove